project_key/models/project_project.py

```diff
@@ -7,7 +7,31 @@
 
 class Project(models.Model):
     _inherit = "project.project"
-    _rec_names_search = ["key", "name", "id"]
+    _rec_names_search = ["key", "name"]
+
+    @api.model
+    def _name_search(
+        self, name="", args=None, operator="ilike", limit=100, name_get_uid=None
+    ):
+        args = list(args or [])
+        search_domain = []
+
+        if name:
+            search_domain = [
+                ("key", operator, name),
+                ("name", operator, name),
+            ]
+            if name.isdigit():
+                search_domain.append(("id", "=", int(name)))
+
+            if len(search_domain) > 1:
+                domain = ["|"] * (len(search_domain) - 1) + search_domain
+            else:
+                domain = search_domain
+        else:
+            domain = []
+
+        return self._search(domain + args, limit=limit, access_rights_uid=name_get_uid)
```

Comment:
issue: IIUC, you don't need to redo everything that `_name_search` will do automatically when `_rec_names_search` is properly configured.

Besides, overriding without calling `super()` is a general bad practice.

Check this please:

```suggestion
        args = list(args or [])
        if name.isdigit():
            args.insert(0, ("id", "=", int(name)))
        return super()._name_search(name, args, operator, limit, name_get_uid)
```